rdrf/rdrf/forms/dynamic/field_lookup.py

```diff
@@ -356,7 +356,10 @@ class DatatypeEWidgetSomeDataTypeName
                     widget = None
 
                 if self.cde.allow_multiple:
-                    widget = widget or CheckboxSelectMultiple
+                    if self.cde.randomise:
+                        widget = widget or widgets.ReadOnlyCheckboxSelectMultiple
```

Comment:
Again why isn't this just?

`widget = widgets.ReadOnlyCheckboxSelectMultiple`

ie. don't allow the widget to be overwritten if the CDE has `randomised` checked. That could just lead to unnecessary complications.